src/components/DeleteComment.jsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from "react";
+import { deleteComment } from "../utils/api";
+
+
+
+const DeleteComment = ({ comment_id, setDeleteStatus }) => {
+
+  const [deleting, setDeleting] = useState(true);
+
+  useEffect(() => {
+    deleteComment(comment_id)
+    .then(() => {
+      setDeleteStatus(false);
+      setDeleting(false);
+      alert('comment deleted successfully!');
+    })
+    .catch((err) => {
+      console.log(err);
+      alert('something went wrong! please try again...');
+      setDeleting(false);
+    });
+
+    if(deleting)
+    alert('request in progress....')
```

Comment:
alerts in js are blocking, so they stop the rest of the code running, this will work as is, but it may be better to have this not be an alert.

Reply:
removed alerts.